diff --git a/public/index.html b/public/index.html
index 47b7dca..2ab9a3b 100644
--- a/public/index.html
+++ b/public/index.html
@@ -23,6 +23,7 @@
     </div>
     
     <div class="result">Hydration = <span id="result">72</span>%
+    <span id="hydration-lock" lock="false">Hydration Lock</span>
     <a href="http://github.com/joshuacronemeyer"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://assets1.github.com/img/71eeaab9d563c2b3c590319b398dd35683265e85?repo=&url=http%3A%2F%2Fs3.amazonaws.com%2Fgithub%2Fribbons%2Fforkme_right_gray_6d6d6d.png&path=" alt="Fork me on GitHub"></a>
     </div>
     
diff --git a/public/js/calculator.js b/public/js/calculator.js
index b69220b..a52e235 100644
--- a/public/js/calculator.js
+++ b/public/js/calculator.js
@@ -27,20 +27,26 @@ $(document).ready(function(){
     water: integerInput('#water-size'),
     starter: integerInput('#starter-size'),
     starterHydration: integerInput('#starter-hydration'),
-
-    addIngredient: function(more, ingredient){
-      return this[ingredient] = this[ingredient] + more;
-    },
-  
+    calculatedHydration: integerInput('#result'),
+    hydrationLock: $('#hydration-lock').attr('lock'),
+    
     calculateHydration: function(){
       var hydration = (((this.water + this.starterWater())/(this.flour + this.starterFlour()))*100);
-      var roundedHydration =  Math.round(hydration);
-      if (isNaN(roundedHydration)){
+      return this.round(hydration);
+    },
+    
+    round: function(decimal){
+      var roundedNumber =  Math.round(decimal);
+      if (isNaN(roundedNumber)){
         return 0;
       }
-      return roundedHydration;
+      return roundedNumber;
     },
-  
+    
+    calculateWeight: function(){
+      return this.flour + this.water + this.starter;
+    },
+    
     starterFlour: function(){
       return (this.starter/ (1+(this.starterHydration/100.0)));
     },
@@ -56,9 +62,37 @@ $(document).ready(function(){
       this.starterHydration = integerInput('#starter-hydration');
     },
     
+    balanceFlour: function(){
+      return this.round((this.water + this.starterWater()) / (this.calculatedHydration/100));
+    },
+    
+    balanceWater: function(){
+      return this.round((this.calculatedHydration/100) * (this.flour + this.starterFlour()));
+    },
+    
     resetHydration: function(){
       this.reload();
       $('#result').text(this.calculateHydration());
+    },
+    
+    addFlour: function(lockedHydration){
+      if (lockedHydration){
+        this.flour = integerInput('#flour-size');
+        $('#water-size').text(this.balanceWater());
+        $("#water").width(this.balanceWater());
+      } else {
+        this.resetHydration();
+      }
+    },
+    
+    addWater: function(lockedHydration){
+      if (lockedHydration){
+        this.water = integerInput('#water-size');
+        $('#flour-size').text(this.balanceFlour());
+        $("#flour").width(this.balanceFlour());
+      } else {
+        this.resetHydration();
+      }
     }
   }
 });
\ No newline at end of file
diff --git a/public/js/ui_magic.js b/public/js/ui_magic.js
index b2e5d5b..f8643cf 100644
--- a/public/js/ui_magic.js
+++ b/public/js/ui_magic.js
@@ -24,6 +24,14 @@ function setStarterHydration(){
   $("#info-result").text($("#result").text());
 }
 
+function hydrationLock(val){
+  if (typeof val == "undefined"){
+    return (/^true$/i).test($('#hydration-lock').attr('lock'));
+  } else {
+    $('#hydration-lock').attr('lock', val);
+  }
+}
+
 baseTwitterIframeSRC = "http://platform0.twitter.com/widgets/tweet_button.html?_=1298156795174&count=vertical&lang=en&url=http%3A%2F%2Fjoshuacronemeyer.github.com%2FFlour-and-Water&via=MakingLoaf"
 function setTweetText(){
   var hydration = $("#result").text();
@@ -32,13 +40,19 @@ function setTweetText(){
   $("iframe").attr("src", baseTwitterIframeSRC + encodedTweetText);
 }
 
-function updateHydration() {
-  $(this).find('.quantity').text($(this).width());
-  calculator.resetHydration();
+function updateWater() {
+  $("#water-size").text($("#water").width());
+  calculator.addWater(hydrationLock());
   $("#info-result").text($("#result").text());
 }
 
-function updateStarterAndHydration() { 
+function updateFlour() {
+  $("#flour-size").text($("#flour").width());
+  calculator.addFlour(hydrationLock());
+  $("#info-result").text($("#result").text());
+}
+
+function updateStarterAndHydration() {
   $(this).find('.quantity').text($(this).width());
   setStarterHydration();
   calculator.resetHydration();
@@ -53,6 +67,13 @@ function resizeableParams(updateCallback){
 	  stop: setTweetText
   }
 }
+
+function toggleHydrationLock(){
+  hydrationLock(!hydrationLock());
+  $("#hydration").resizable( "option", "disabled", hydrationLock() );
+  $("#starter").resizable( "option", "disabled", hydrationLock() );
+}
+
 $(function() {
   setTweetText();
   $("#info").click(function(){$("#overlay").show();});
@@ -61,8 +82,9 @@ $(function() {
       $("#overlay").hide();
     }
   });
-  $("#flour").resizable(resizeableParams(updateHydration));
-  $("#water").resizable(resizeableParams(updateHydration));
+  $("#flour").resizable(resizeableParams(updateFlour));
+  $("#water").resizable(resizeableParams(updateWater));
   $("#starter").resizable($.extend(resizeableParams(updateStarterAndHydration), {alsoResize: "#hydration"}));
 	$("#hydration").resizable($.extend(resizeableParams(setStarterHydration), {containment: "#starter"}));
+	$("#hydration-lock").click(toggleHydrationLock);
 });
\ No newline at end of file
diff --git a/test/flour_n_water_test.html b/test/flour_n_water_test.html
index 7202779..54e6072 100644
--- a/test/flour_n_water_test.html
+++ b/test/flour_n_water_test.html
@@ -15,6 +15,10 @@
       $('#starter-hydration').text(hydration);
     }
     
+    function setHydrationLock(){
+      $('#hydration-lock').attr('lock', true);
+    }
+    
     $(document).ready(function(){
     
       test("calculate 50% hydration plus 50% hydration starter", function(){
@@ -55,6 +59,26 @@
         equals($('#result').text(), 50, "We expect 50 percent hydration" );
       });
       
+      test("calculate dough weight", function() {
+        setAmounts(100, 100, 100, 100);
+        calculator.reload();
+        equals(calculator.calculateWeight(), 300, "We expect 300 grams" );
+      });
+      
+      // test("locking hydration scales dough weight", function() {
+      //   var flour = 100;
+      //   setAmounts(flour, 100, 100, 100);
+      //   calculator.reload();
+      //   equals(calculator.calculateHydration(), 100, "We expect 100 percent hydration" );
+      //   equals(calculator.calculateWeight(), 300, "We expect 300 grams" );
+      //   setHydrationLock();
+      //   var additionalFlour = 100;
+      //   setAmounts(flour + additionalFlour, 100, 100, 100);
+      //   calculator.reload();
+      //   equals(calculator.calculateHydration(), 100, "We expect 100 percent hydration" );
+      //   equals(calculator.calculateWeight(), 500, "We expect 500 grams" );
+      // });
+      // 
     });
   </script> 
 </head>
@@ -70,6 +94,7 @@
     <span id="water-size">250</span>
     <span id="starter-size">250</span>
     <span id="starter-hydration">100</span>
+    <span id="hydration-lock" lock='false'></span>
     <span id="result">0</span>
   </div>
 </body>
